database/test_database.py

```diff
@@ -0,0 +1,103 @@
+from unittest import TestCase
+
+from database.image import *
+from database.project import *
+
+
+# Before running make sure no .db files are present in execution directory
+
+
+# Tests for Project Class
+class TestDatabase(TestCase):
+    # should not actually be called
+    # the test for insert
+    # insert two rows and verify with fetch
+    # also verify for row which is not there (testing fetch also)
+    def insert_project_test(self):
+        # insert a project
+        pr = ProjectRepository()
+        pr.insert("project 1", "network 1")
+
+        # insert another project
+        pr.insert("project 2", "network 2")
+
+        # check whether first project is present
+        pid = pr.fetch_id_with_name("project 1")
+        self.assertIsNotNone(pid)
+        self.assertEqual(pid, 1)
+
+        # check whether second project is present
+        pid = pr.fetch_id_with_name("project 2")
+        self.assertIsNotNone(pid)
+        self.assertEqual(pid, 2)
+
+        # verify that None is being returned for a project which doesnt exist
+        pid = pr.fetch_id_with_name("project 3")
+        self.assertIsNone(pid)
+
+    # should not actually be called
+    # the test for delete
+    # delete the inserted row
+    # check if it is gone
```

Comment:
The comments don't really add any value. Please only use comments when you feel it is necessary to explain something not clear in code.